contracts/aggregator/fantom/flashloan/implementations/implAaveV3/main.sol

```diff
@@ -0,0 +1,128 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+import '../../helper.sol';
+
+interface IAaveV3Lending {
+    function flashLoan(
+        address receiverAddress,
+        address[] calldata assets,
+        uint256[] calldata amounts,
+        uint256[] calldata modes,
+        address onBehalfOf,
+        bytes calldata params,
+        uint16 referralCode
+    ) external;
+
+    function FLASHLOAN_PREMIUM_TOTAL() external view returns (uint128);
+}
+
+contract Variables {
+    address public constant aaveV3LendingAddr = 0x794a61358D6845594F94dc1DB02A252b5b4814aD;
+    IAaveV3Lending public constant aaveV3Lending = IAaveV3Lending(aaveV3LendingAddr);
+}
+
+contract AaveImplementationFantom is Helper, Variables {
+    /**
+     * @dev Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @notice Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @param _tokens token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _route route for flashloan.
+     * @param _data extra data passed.
+     */
+    function flashLoan(
+        address[] memory _tokens,
+        uint256[] memory _amounts,
+        uint256 _route,
+        bytes calldata _data,
+        bytes calldata // kept for future use by instadapp. Currently not used anywhere.
+    ) external reentrancy {
+        require(_route == 9, 'invalid-AAVE-route');
+        (_tokens, _amounts) = bubbleSort(_tokens, _amounts);
+        validateTokens(_tokens);
+        routeAaveV3(_tokens, _amounts, _data);
+    }
+
+    /**
+     * @dev Returns fee for the passed route in BPS.
+     * @notice Returns fee for the passed route in BPS. 1 BPS == 0.01%.
+     * @param _route route number for flashloan.
+     */
+    function calculateFeeBPS(uint256 _route) public view returns (uint256 BPS_) {
+        if (_route == 9) {
+            BPS_ = aaveV3Lending.FLASHLOAN_PREMIUM_TOTAL();
+        } else {
+            revert('Invalid source');
+        }
```

Comment:
Where is the logic where we take 5 BPS if the fee is less. We need it here because we are using this function directly in the `executeOperation` function

Reply:
Hey, we have this logic here - [calculateFeeBPS](https://github.com/Instadapp/flashloan-aggregator/blob/Fantom-support-FLA/contracts/aggregator/common/main.sol#L111), if the bps is less, it will take InstaFeeBPS. Instead of adding it to all the implementations, it's getting checked at the end from the common function.

Reply:
@shriyatyagii But it needs to be in the implementations because its used there to calculate `instafees`

Reply:
ah! Gotcha! Updated